env/env_test.cc

```diff
@@ -3110,6 +3110,88 @@ TEST_F(EnvTest, CreateCompositeEnv) {
 }
 #endif  // ROCKSDB_LITE
 
+class TestAsyncRead : public testing::Test {
+ public:
+  TestAsyncRead() { env_ = Env::Default(); }
+
+  Env* env_;
+};
+
+// Tests the default implementation of ReadAsync API.
+TEST_F(TestAsyncRead, ReadAsync) {
+  EnvOptions soptions;
+  std::string fname = test::PerThreadDBPath(env_, "testfile");
+
+  const size_t kSectorSize = 4096;
+  const size_t kNumSectors = 8;
+
+  // 1. create & write to a file.
+  {
+    std::unique_ptr<FSWritableFile> wfile;
+    ASSERT_OK(Env::Default()->GetFileSystem()->NewWritableFile(
+        fname, FileOptions(), &wfile, nullptr /*dbg*/));
+
+    for (size_t i = 0; i < kNumSectors; ++i) {
+      auto data = NewAligned(kSectorSize * 8, static_cast<char>(i + 1));
+      Slice slice(data.get(), kSectorSize);
+      ASSERT_OK(wfile->Append(slice, IOOptions(), nullptr));
+    }
+    ASSERT_OK(wfile->Close(IOOptions(), nullptr));
+  }
+  // 2. Read file.
+  {
+    std::unique_ptr<FSRandomAccessFile> file;
+    ASSERT_OK(Env::Default()->GetFileSystem()->NewRandomAccessFile(
```

Comment:
I was hoping we could have a mock file system (derived from FileSystemWrapper) and a mock random access file (derived from FSRandomAccessFileWrapper). It can be very simple, just override NewRandomAccessFile and ReadAsync respectively.